src/extensions/data_array.js

```diff
@@ -67,24 +96,101 @@ export default () => {
       });
     };
 
+    const createInstances = observationObjects => {
+      return new Promise(resolve => {
```

Comment:
You don't need to wrap everything inside a `new Promise`, just do `return db().then(...)` and use `Promise.resolve()` or `Promise.reject()` inside the `then` handler.